### PR TITLE
[ignition-cmake2] Add new port 🤖

### PR DESCRIPTION
**Pull request by traversaro**
Add a port for ignition-cmake2. As discussed in https://github.com/microsoft/vcpkg/pull/7781, different major version of ignition robotics libraries (https://ignitionrobotics.org/) can be installed side by side, so each new major version is added as a new port. 

In particular, ignition-cmake2 is a dependency for Gazebo 11 (http://gazebosim.org/blog/gazebo11) and for Ignition Robotics Citadel (that contains Ignition Gazebo 3, see https://www.openrobotics.org/blog/2019/12/11/ignition-citadel-released).

As far as I was able to test, this port should support all the triplets.

Related issue:
* https://github.com/microsoft/vcpkg/issues/8014#issuecomment-578431035

### CODE REVIEW

File: ports/ignition-cmake2/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,18 @@
+include(vcpkg_common_functions)
```
Comment (NancyLi1013) [inline ports/ignition-cmake2/portfile.cmake]: This function is no longer needed.
Could you please remove it?

File: ports/ignition-modularscripts/CONTROL [hunk 2]
```diff
@@ -1,3 +1,3 @@
 Source: ignition-modularscripts
-Version: 2019-09-11
+Version: 2020-02-10
 Description: Vcpkg helpers to package ignition libraries
```
Comment (NancyLi1013) [inline ports/ignition-modularscripts/CONTROL]: Could you please add homepage here?

File: ports/ignition-modularscripts/ignition_modular_library.cmake [hunk 3]
```diff
@@ -8,19 +8,21 @@ function(ignition_modular_build_library NAME MAJOR_VERSION SOURCE_PATH CMAKE_PAC
...
+                  "${CURRENT_PACKAGES_DIR}/share/${CMAKE_PACKAGE_NAME}/")
 
+        # Remove debug files
+        file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include
+                            ${CURRENT_PACKAGES_DIR}/debug/lib/cmake
+                            ${CURRENT_PACKAGES_DIR}/debug/share)
+    endif()
 
 
     # Post-build test for cmake libraries
```
Comment (NancyLi1013) [inline ports/ignition-modularscripts/ignition_modular_library.cmake]: ` vcpkg_test_cmake()` is deprecated. Could you please remove it?

File: ports/ignition-cmake2/CONTROL [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Source: ignition-cmake2
+Version: 2.1.1
+Homepage: https://ignitionrobotics.org/libs/cmake
+Description: CMake helper functions for building robotic applications
+Build-Depends: ignition-modularscripts
```
Comment (NancyLi1013) [inline ports/ignition-cmake2/CONTROL]: The following space lines should be removed.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/ignition-modularscripts/CONTROL (hunk 2 above)
Comment (traversaro) [reply ports/ignition-modularscripts/CONTROL]: This is a vcpkg-only port of scripts that help package ignition libraries, but it is not code that is provided by the ignition project at all, so I think adding an homepage would be misleading. 

Furthermore, similar ports for boost and  qt do not provide an homepage: 
* https://github.com/microsoft/vcpkg/blob/master/ports/qt5-modularscripts/CONTROL
* https://github.com/microsoft/vcpkg/blob/master/ports/boost-modular-build-helper/CONTROL

If a homepage is strictly required, we can add either the vcpkg one ( https://github.com/microsoft/vcpkg )  or the ignition libraries one ( https://ignitionrobotics.org/).

File: ports/ignition-cmake2/CONTROL (hunk 4 above)
Comment (traversaro) [reply ports/ignition-cmake2/CONTROL]: Done.

Comment (traversaro) [reply ports/ignition-cmake2/CONTROL]: Done.

File: ports/ignition-modularscripts/ignition_modular_library.cmake (hunk 3 above)
Comment (traversaro) [reply ports/ignition-modularscripts/ignition_modular_library.cmake]: Done.

**Comment by traversaro** [comment]
The CI now fails due to a qt5-tools regression, but I am not sure how this is related to the PR.

**Comment by NancyLi1013** [comment]
It seems that there is something wrong with Linux platform. 
The regression should be not related to this PR.
I have re run it. Please wait for some time.

File: ports/ignition-cmake2/portfile.cmake (hunk 1 above)
Comment (traversaro) [reply ports/ignition-cmake2/portfile.cmake]: Done.

**Comment by traversaro** [comment]
Hi @NancyLi1013 , I think I addressed all your comments, so I am not sure why the PR is listed under  "waiting for response"  . Let me know if I need to do something, thanks!

**Comment by NancyLi1013** [comment]
Hi @traversaro 

Sorry for the late response.
Since the CI test results were not all passed when I checked them before.

File: ports/ignition-modularscripts/CONTROL (hunk 2 above)
Comment (JackBoosY) [reply ports/ignition-modularscripts/CONTROL]: No need to add homepage in this case.

File: ports/ignition-cmake2/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+                         SHA512 4d22a45ccc9582c7e4b370b884511782d1629fa3e257dd92300388b5050d22fa63dd4a6ef8942abb9ebbc300df4cd526d1d8a7088a92b0073e152c16c7b97e2b)
+
+# Permit empty include folder
+set(VCPKG_POLICY_EMPTY_INCLUDE_FOLDER enabled)
+
+# Remove unneccessary directory, as ignition-cmake is a pure CMake package
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib ${CURRENT_PACKAGES_DIR}/debug)
+
+# Install custom usage
+configure_file(${CMAKE_CURRENT_LIST_DIR}/usage ${CURRENT_PACKAGES_DIR}/share/${PORT}/usage @ONLY)
```
Comment (JackBoosY) [inline ports/ignition-cmake2/portfile.cmake]: Miss copyright?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/ignition-cmake2/portfile.cmake (hunk 5 above)
Comment (traversaro) [reply ports/ignition-cmake2/portfile.cmake]: This is handled by `ignition_modular_library` CMake function in https://github.com/microsoft/vcpkg/blob/ca6e6d09ce1cbf956e94ca01079ad3f778377a8c/ports/ignition-modularscripts/ignition_modular_library.cmake#L33 .

**Review by JackBoosY — APPROVED** [review]

**Comment by strega-nil** [comment]
Alright, looks good to me!

**Comment by traversaro** [comment]
Thanks a lot @strega-nil @NancyLi1013 @JackBoosY !

**Pull request merged**